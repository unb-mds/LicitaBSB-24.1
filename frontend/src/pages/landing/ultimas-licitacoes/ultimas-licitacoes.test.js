import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import UltimasLicitacoes from '.';


describe('UltimasLicitacoes', () => {
    test('renderizar o título do componente', () => {
        render(<MemoryRouter>
            <UltimasLicitacoes />
        </MemoryRouter>)
        const title = screen.getByRole('heading')
        expect(title).toBeInTheDocument()
    })
    test('renderizar o link para página Ver mais licitações', () => {
        render(<MemoryRouter>
            <UltimasLicitacoes />
        </MemoryRouter>)
        const link = screen.getByTestId('link-testid')
        expect(link).toBeInTheDocument()
    })
    test('deve a lista com as licitações mais recentes', () => {
        render(<MemoryRouter>
            <UltimasLicitacoes />
        </MemoryRouter>)
        const link = screen.getByRole('list')
        expect(link).toBeInTheDocument()
    });
    test('o snapshot com a lista de licitações', () => {
        render(<MemoryRouter>
            <UltimasLicitacoes />
        </MemoryRouter>)
        const link = screen.getByTestId('main-testid')
        expect(link).toMatchSnapshot()
    })
});