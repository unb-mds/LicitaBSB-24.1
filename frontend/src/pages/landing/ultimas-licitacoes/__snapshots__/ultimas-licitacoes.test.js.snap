// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UltimasLicitacoes o snapshot com a lista de licitações 1`] = `
<div
  class="wrapper"
  data-testid="main-testid"
>
  <div
    class="licitacoesHeader"
  >
    <h2>
      Últimas Licitações
    </h2>
    <a
      class="linkLicitacoes"
      data-testid="link-testid"
      href="/licitacoes"
    >
      Ver todas as licitações
    </a>
  </div>
  <ul
    class="licitacoesWrapper"
  >
    <li
      data-testid="listitem-testid"
    >
      <div
        class="cardWrapper"
      >
        <h5
          class="cardTitle"
        >
          Órgão A
        </h5>
        <div>
          <div
            class="cardStatus"
          >
            <div
              class="statusContainer"
            >
              <p
                class="cardStatusText"
                data-testid="status-testid"
              >
                Status: 
                Aberta
              </p>
            </div>
            <p
              class="cardStatusText"
              data-testid="modalidade-testid"
            >
              Modalidade: 
              Tipo A
            </p>
          </div>
          <div
            class="licitacoesInfo"
          >
            <p
              data-testid="data-testid"
            >
              Data de publicação: 
              01/01/2024
            </p>
          </div>
          <div
            class="cardSection"
          />
          <div>
            <p
              class="cardDescricao"
            />
          </div>
        </div>
        <div>
          <a
            class="cardButton"
            data-testid="link-testid"
          >
            <p>
              Ver Mais
            </p>
          </a>
        </div>
      </div>
    </li>
    <li
      data-testid="listitem-testid"
    >
      <div
        class="cardWrapper"
      >
        <h5
          class="cardTitle"
        >
          Órgão B
        </h5>
        <div>
          <div
            class="cardStatus"
          >
            <div
              class="statusContainer"
            >
              <p
                class="cardStatusText"
                data-testid="status-testid"
              >
                Status: 
                Aberta
              </p>
            </div>
            <p
              class="cardStatusText"
              data-testid="modalidade-testid"
            >
              Modalidade: 
              Tipo B
            </p>
          </div>
          <div
            class="licitacoesInfo"
          >
            <p
              data-testid="data-testid"
            >
              Data de publicação: 
              02/01/2024
            </p>
          </div>
          <div
            class="cardSection"
          />
          <div>
            <p
              class="cardDescricao"
            />
          </div>
        </div>
        <div>
          <a
            class="cardButton"
            data-testid="link-testid"
          >
            <p>
              Ver Mais
            </p>
          </a>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
